File: backend/app/schemas/sessions.py
# Copyright (C) 2026 LIHUO. All rights reserved.
#
# This file is released under the MIT License.

from pydantic import BaseModel, Field


class SessionCreateRequest(BaseModel):
    client_user_id: str = "anonymous"
    title: str | None = None
    asset_ids: list[str] = Field(default_factory=list)


class SessionSummary(BaseModel):
    session_id: str
    title: str | None
    status: str
    last_message: str | None = None
    subject: str | None = None
    knowledge_points: list[str] = Field(default_factory=list)


class SessionListResponse(BaseModel):
    items: list[SessionSummary]


class SessionDetailResponse(SessionSummary):
    client_user_id: str
    asset_ids: list[str] = Field(default_factory=list)
    messages: list[dict] = Field(default_factory=list)

File: frontend/src/api/sessions.ts
/**
 * 会话相关 API
 */
import { request, clientUserId } from '../utils/request'

export interface Session {
  session_id: string
  title: string
  status: string
  client_user_id: string
  asset_ids: string[]
  messages: Message[]
}

export interface Message {
  message_id: string
  role: 'user' | 'assistant'
  content: string
  created_at: string
}

export interface SessionListItem {
  session_id: string
  title: string
  status: string
  last_message: string
  subject: string
  knowledge_points: string[]
}

export function createSession(params?: { title?: string; asset_ids?: string[] }) {
  return request<Session>('/api/v1/sessions', {
    method: 'POST',
    data: {
      client_user_id: clientUserId,
      ...params,
    },
  })
}

export function getSessions() {
  return request<{ items: SessionListItem[] }>(
    `/api/v1/sessions?client_user_id=${encodeURIComponent(clientUserId)}`,
  )
}

export function getSession(sessionId: string) {
  return request<Session>(`/api/v1/sessions/${sessionId}`)
}
